Permission API updates (#35768)
mdn/content: files/en-us/web/api/geolocation_api/index.md

- {{domxref("Navigator.geolocation")}}
  - : The entry point into the API. Returns a {{domxref("Geolocation")}} object instance, from which all other functionality can be accessed.

> ## Security considerations

Author: Decided to add while I am here. Would like to add for all thinks with a user permission, including the specific API calls that invoke the prompt.

The Geolocation API allows users to programmatically access location information in [secure contexts](/en-US/docs/Web/Security/Secure_Contexts).

Access may further be controlled by the [Permissions Policy](/en-US/docs/Web/HTTP/Permissions_Policy) directive {{HTTPHeader("Permissions-Policy/geolocation","geolocation")}}.

> The default allowlist for `geolocation` is `self`, which allows access to position information only same-origin nested frames.

Reviewer: what do you think?
```suggestion
The default allowlist for `geolocation` is `self`, which allows access to location information in same-origin nested frames only.
```

> Third party usage can be enabled by the server first setting the `Permissions-Policy` header to grant permission a particular third party origin:

Reviewer: Is this better?

```suggestion
Third party usage is enabled by setting a `Permissions-Policy` response header to grant permission to a particular third party origin:
```

```http
Permissions-Policy: geolocation=(self b.example.com)
```

> Then the `allow="geolocation"` attribute must be added the frame container element for sources from that origin:

Reviewer: And here?

```suggestion
The `allow="geolocation"` attribute must then be added to the iframe element with sources from that origin:
```

```html
<iframe src="https://b.example.com" allow="geolocation"/></iframe>
```

> Geolocation is a powerful feature, and end users must additionally grant explicit permission via a prompt when either {{domxref("Geolocation.getCurrentPosition()")}} or {{domxref("Geolocation.watchPosition()")}} is called (unless the permission state is already `granted` or `denied`).

Reviewer: Just some more detail why it's powerful

```suggestion
Geolocation data may be sensitive, revealing, or unwanted by visitors.
Therefore, users must grant explicit permission via a prompt when either {{domxref("Geolocation.getCurrentPosition()")}} or {{domxref("Geolocation.watchPosition()")}} is called (unless the permission state is already `granted` or `denied`).
```

Author: I kind of liked the term "powerful feature", simply because when you see that in the spec it always means "requires a user prompt". On the other hand, explaining why it is powerful is more useful. I'm not sure about "unwanted by visitors", so a slight modification.

```suggestion
Geolocation data may reveal information that the device owner does not want to share.
Therefore, users must grant explicit permission via a prompt when either {{domxref("Geolocation.getCurrentPosition()")}} or {{domxref("Geolocation.watchPosition()")}} is called (unless the permission state is already `granted` or `denied`).
```
